test/rules/parameter_assignments.dart

```diff
@@ -45,3 +49,34 @@ void otherBad1(int parameter) { // LINT
   parameter += 3;
```

Comment:
This is because we are directly reporting required parameters before traversing the nodes mutating parameters. We just rely on the fact that isPotentiallyMutatedInScope methods are returning true.